optics-core/src/Optics/Internal/Optic/Subtyping.hs

```diff
@@ -26,17 +27,71 @@ class Is k l where
   -- | Witness of the subtyping relationship.
   implies :: (Constraints k p => r) -> (Constraints l p => r)
 
--- | Overlappable instance for a custom type error.
-instance {-# OVERLAPPABLE #-} TypeError ('ShowType k
-                                         ':<>: 'Text " cannot be used as "
-                                         ':<>: 'ShowType l
-                                        ) => Is k l where
-  implies _ = error "unreachable"
-
 -- | Every kind of optic can be used as itself.
 instance Is k k where
   implies r = r
 
+-- | Overlappable instance for a custom type error.
+instance {-# OVERLAPPABLE #-} TypeError
+  ('ShowType k ':<>: 'Text " cannot be used as " ':<>: 'ShowType l
+   ':$$: 'Text "Perhaps you meant to use one of these:"
+   ':$$: EliminationForms k
+  ) => Is k l where
+  implies _ = error "unreachable"
+
+type family EliminationForms (k :: OpticKind) :: ErrorMessage where
+  EliminationForms An_AffineFold      = AffineFoldEliminations
+  EliminationForms An_AffineTraversal = AffineTraversalEliminations
+  EliminationForms A_Fold             = FoldEliminations
+  EliminationForms A_Getter           = GetterEliminations
+  EliminationForms A_Lens             = LensEliminations
+  EliminationForms A_Prism            = PrismEliminations
+  EliminationForms A_ReversedLens     = PrismEliminations
+  EliminationForms A_ReversedPrism    = LensEliminations
+  EliminationForms A_Review           = ReviewEliminations
+  EliminationForms A_Setter           = SetterEliminations
+  EliminationForms A_Traversal        = TraversalEliminations
+
+type AffineFoldEliminations
+  =     ShowSymbolWithOrigin "preview" "Optics.AffineFold"
+  ':$$: ShowSymbolWithOrigin "(^?)"    "Optics.Operators"
+
+type AffineTraversalEliminations
+  =     AffineFoldEliminations
+  ':$$: SetterEliminations
+
+type FoldEliminations
+  =     ShowSymbolWithOrigin "traverseOf_" "Optics.Fold"
+  ':$$: ShowSymbolWithOrigin "foldMapOf"   "Optics.Fold"
+  ':$$: ShowSymbolWithOrigin "toListOf"    "Optics.Fold"
+  ':$$: ShowSymbolWithOrigin "(^..)"       "Optics.Operators"
+
+type GetterEliminations
+  =     ShowSymbolWithOrigin "view" "Optics.Getter"
+  ':$$: ShowSymbolWithOrigin "(^.)" "Optics.Operators"
+
+type LensEliminations
+  =     GetterEliminations
+  ':$$: SetterEliminations
+
+type PrismEliminations
+  =     AffineFoldEliminations
+  ':$$: ReviewEliminations
+
+type ReviewEliminations
+  =     ShowSymbolWithOrigin "review" "Optics.Review"
+  ':$$: ShowSymbolWithOrigin "(#)"    "Optics.Operators"
+
+type SetterEliminations
+  =     ShowSymbolWithOrigin "over" "Optics.Setter"
+  ':$$: ShowSymbolWithOrigin "(%~)" "Optics.Operators"
```

Comment:
I find it's slightly messy, would be nicer to have named combinators as they are now, but all operators as a single list.

I.e not

```
      Perhaps you meant to use one of these:
        ‘preview’ (from Optics.AffineFold)
        ‘(^?)’ (from Optics.Operators)
        ‘over’ (from Optics.Setter)
        ‘(%~)’ (from Optics.Operators)
        ‘set’ (from Optics.Setter)
        ‘(.~)’ (from Optics.Operators)
```

but

```
      Perhaps you meant to use one of these:
        ‘preview’ (from Optics.AffineFold)
        ‘over’ (from Optics.Setter)
        ‘set’ (from Optics.Setter)
        ‘(^?)’ ‘(%~)’ ‘(.~)’ (from Optics.Operators)
```

How hard would to do that change?

(I wouldn't group named combinators by module, but that's an option too if it makes stuff easier).